typescript/sdk/src/ism/adapters/SealevelMultisigAdapter.ts

```diff
@@ -0,0 +1,196 @@
+      // 1. `[]` The access control PDA account.
+      { pubkey: accessControlPda, isSigner: false, isWritable: false },
+      // 2. `[writable]` The PDA relating to the provided domain.
+      { pubkey: domainDataPda, isSigner: false, isWritable: true },
+      // 3. `[executable]` OPTIONAL - The system program account. Required if creating the domain PDA.
+      { pubkey: SystemProgram.programId, isSigner: false, isWritable: false },
+    ];
+
+    const value = new SealevelInstructionWrapper({
+      instruction:
+        SealevelMultisigIsmInstructionType.SET_VALIDATORS_AND_THRESHOLD,
+      data: new SealevelMultisigIsmSetValidatorsInstruction({
+        domain,
+        validators,
+        threshold,
+      }),
+    });
+
+    const data = Buffer.from(
+      serialize(SealevelMultisigIsmSetValidatorsInstructionSchema, value),
+    );
+
+    return new TransactionInstruction({
+      keys,
+      programId: multisigIsmProgramId,
+      data,
+    });
+  }
+
+  /**
+   * Create a TransferOwnership instruction
+   * @param multisigIsmProgramId - The MultisigIsm program ID
+   * @param accessControlPda - The access control PDA account
+   * @param owner - The current owner
+   * @param newOwner - The new owner (null to renounce ownership)
+   * @returns TransactionInstruction
+   */
+  createTransferOwnershipInstruction(
+    multisigIsmProgramId: PublicKey,
+    accessControlPda: PublicKey,
+    owner: PublicKey,
+    newOwner: PublicKey | null,
+  ): TransactionInstruction {
+    const keys: AccountMeta[] = [
+      // 0. `[signer]` The current access control owner.
+      { pubkey: owner, isSigner: true, isWritable: true },
+      // 1. `[writeable]` The access control PDA account.
+      { pubkey: accessControlPda, isSigner: false, isWritable: true },
+    ];
+
+    const value = new SealevelInstructionWrapper({
+      instruction: SealevelMultisigIsmInstructionType.TRANSFER_OWNERSHIP,
+      data: new SealevelMultisigIsmTransferOwnershipInstruction({
+        newOwner: newOwner ? newOwner.toBuffer() : null,
+      }),
+    });
+
+    const data = Buffer.from(
+      serialize(SealevelMultisigIsmTransferOwnershipInstructionSchema, value),
+    );
+
+    return new TransactionInstruction({
+      keys,
+      programId: multisigIsmProgramId,
+      data,
+    });
+  }
+
+  /**
+   * Helper method to convert hex string validators to Uint8Array format
+   * @param hexValidators - Array of validator addresses as hex strings (with or without 0x prefix)
+   * @returns Array of 20-byte Uint8Arrays
+   */
+  static hexValidatorsToUint8Array(hexValidators: string[]): Uint8Array[] {
+    return hexValidators.map((hex) => {
+      const cleaned = hex.startsWith('0x') ? hex.slice(2) : hex;
+      if (cleaned.length !== 40) {
+        throw new Error(
+          `Validator address must be 40 hex characters (20 bytes), got ${cleaned.length}`,
+        );
+      }
+      return Uint8Array.from(Buffer.from(cleaned, 'hex'));
+    });
+  }
```

Comment:
nit: I think this would be helpful to have in the utils package

Reply:
i thought i'd keep here for now til we have a more general plan for svm - but agreed it's more of a utils thing